chapter_4/ex02_static/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+# CMakeLists file for the Chapter 4 Example nr. 02: 
+# Installable static library
+#
+# SPDX-License-Identifier: MIT
+
+project(
+  ch4_ex02_static
+  VERSION 1.0
+  DESCRIPTION "Chapter 4 Example 02, installable static library project"
+  LANGUAGES CXX)
+
+# Define an executable target named `ch4_ex02_static` 
+add_library(ch4_ex02_static STATIC)
+# Specify source files for target named `ch4_ex02_static`
+target_sources(ch4_ex02_static PRIVATE src/lib.cpp)
+# Specify the include directories for the target named `ch4_ex02_static`
+target_include_directories(ch4_ex02_static PUBLIC include)
+# Specify the public headers for the target named `ch4_ex02_static`
+# By default, CMake does not install any extra files other than output artifacts
+# for the target. In order to install the header files too, target's PUBLIC_HEADER property 
+# must be populated
+set_target_properties(ch4_ex02_static PROPERTIES PUBLIC_HEADER include/chapter4/ex02/lib.hpp)
```

Comment:
`PUBLIC_HEADER` is only relevant for MacOS specific `FRAMEWORK` shared library targets. [Docs](https://cmake.org/cmake/help/latest/prop_tgt/PUBLIC_HEADER.html).

Reply:
I know, but making this property explicit to FRAMEWORK targets makes no sense. I've used this not to have two separate `install(...)` invocations for installing the target.

 CMake should have a way of associating header files with targets. Using PUBLIC_HEADERS property exactly provides that. Looks like I'm not the only one thinking in that way: https://gitlab.kitware.com/cmake/cmake/-/issues/22468

I'll be glad to see an official way of associating headers with targets. The issue I've mentioned is targeted for 3.22 release, so if the proposed feature is delivered before the book hits the shelf, I'll use the proposed `target_headers` instead. Until then, it seems I have to convert this to two separate install calls.

Reply:
3.22 is out the door already and the linked MR is closed for further design discussions, so it's too early to think about using it.

While using `PUBLIC_HEADER` works for sure, however it fails to properly support the most common structure of header files, which is using directories. In fact, the the include directory in the project should have the same structure as when installed, so the `#include`s work everywhere. The includes should also be prefixed to a directory, to minimize the possibility of name clashes with other dependencies or system dependencies. Think how a lot of headers on Linux are prefixed with the `sys/` directory.

Reply:
If I may chime in, I support @friendlyanon s point of view. IMO the clean way to do it (except for MacOS `FRAMEWORK`) is using directories. But mentioning the `PUBLIC_HEADER` property for that case in chapter 4 definitively makes sense.

I wasn't aware of the discussion about PUBLIC_HEADER, but after reading this I got the impression that it is not yet agreed on how to proceed here.